spec/system/exercises_system_spec.rb

```diff
@@ -0,0 +1,162 @@
+    click_button(I18n.t('sessions.new.link'))
+    wait_for_ajax
+  end
+
+  context 'when an exercise created' do
+    let(:teacher) { create(:teacher) }
+    let(:submission_deadline) { 3.months.from_now.beginning_of_minute }
+    let(:late_submission_deadline) { submission_deadline + 1.week }
+    let(:title) { 'Ruby challenge' }
+    let(:internal_title) { 'Project Ruby: Table flip' }
+
+    let(:code) do
+      <<~RUBY
+        def self.❨╯°□°❩╯︵┻━┻
+          puts "Calm down, bro"
+        end
+      RUBY
+    end
+
+    let(:description) do
+      <<~TEXT
+        Ruby challenge
+
+        Do something with Ruby.
+      TEXT
+    end
+
+    before do
+      visit(exercises_path)
+      click_on I18n.t('shared.new_model', model: Exercise.model_name.human)
+      wait_for_ajax
+    end
+
+    it 'creates an exercise with nested data' do
+      fill_in Exercise.human_attribute_name(:title), with: title
+      fill_in Exercise.human_attribute_name(:internal_title), with: internal_title
+
+      # description
+      within('.markdown-editor__wrapper') do
+        find('.ProseMirror').set(description)
+      end
+
+      chosen_select(Exercise.human_attribute_name(:execution_environment), ruby.name)
+
+      chosen_date_time_select(Exercise.human_attribute_name(:submission_deadline), submission_deadline)
+
+      chosen_date_time_select(Exercise.human_attribute_name(:late_submission_deadline), late_submission_deadline)
+
+      check Exercise.human_attribute_name(:public)
+
+      click_on I18n.t('exercises.form.add_file')
+
+      within(find_by_id('files').all('li').last) do
+        fill_in CodeOcean::File.human_attribute_name(:name), with: 'main'
+
+        chosen_select(CodeOcean::File.human_attribute_name(:file_type), ruby.file_type.name)
+        chosen_select(CodeOcean::File.human_attribute_name(:role), I18n.t('code_ocean/files.roles.main_file'))
+
+        check(CodeOcean::File.human_attribute_name(:read_only))
+
+        find_by_id('editor-edit').click
+        send_keys code.strip
+      end
+
+      click_button I18n.t('shared.create', model: Exercise.model_name.human)
+
+      expect(page).to have_text 'Exercise has successfully been created.'
+
+      # Exercise is created with expected attributes
+      expect(page).to have_text(title)
+      expect(page).to have_text(internal_title)
+      expect(page).to have_text(submission_deadline.to_s)
+      expect(page).to have_text(late_submission_deadline.to_s)
+      expect(page).to have_text(ruby.name)
+
+      description.lines.each do |line|
+        expect(page).to have_text(line.delete("\n"))
+      end
+
+      # Exercise includes the code
+      find('span', text: 'main.rb').click
+
+      code.lines.each do |code_line|
+        expect(page).to have_text(code_line.delete("\n"))
+      end
+    end
+  end
+
+  context 'when an exercise is updated' do
+    let!(:exercise) { create(:fibonacci) }
+    let(:teacher) { exercise.user }
```

Comment:
```suggestion
    let!(:exercise) { create(:fibonacci, user: teacher) }
```

Reply:
I would move the teacher declaration to the outer scope (and assign the user in the exercise directly)